### ISSUE TITLE
pyarrow.lib.ArrowCapacityError

### ISSUE THREAD

**Issue by conceptofmind** [Eventual-Inc/Daft#3049]
### Describe the bug

```
    _filesystemdataset_write(
  File "pyarrow/_dataset.pyx", line 4010, in pyarrow._dataset._filesystemdataset_write
  File "pyarrow/error.pxi", line 92, in pyarrow.lib.check_status
pyarrow.lib.ArrowCapacityError: array cannot contain more than 2147483646 bytes, have 2230017098
```

### To Reproduce

```python
df_img_bytes.write_parquet(f'my-data/')
```

### Expected behavior

Write the bytes to parquet.

### Component(s)

Parquet

### Additional context

It looks like if writing a large array the `to_parquet` function will crash.

**Comment by jaychia** [comment]
Thanks @conceptofmind ! We're putting this on the list of priorities to investigate.

**Comment by jaychia** [comment]
Yes Arrow really doesn't like it when it encounters byte arrays more than 2GB in total length (this is one reason why storing encoded images into Parquet directly is a bad idea... Lots of Arrow problems)

**Comment by conceptofmind** [comment]
Same thing always happens with long context text data :laughing:

**Comment by MisterKloudy** [comment]
Yes I've definitely seen this before as well for pure text data.
